config/opal_check_ofi.m4

```diff
@@ -81,6 +135,30 @@ AC_DEFUN([_OPAL_CHECK_OFI],[
                               [],
                               [opal_ofi_happy=no])])
 
+    dnl If we got happiness, then check the Libfabric API version.
+    AS_IF([test "$opal_ofi_happy" = "yes"], [
+        AC_PROG_CPP
+
+        AC_MSG_CHECKING([for major/minor OFI API version])
+
+        dnl The goal here is to do this test without compiling/runnning
+        dnl a program (because that is problematic in cross-compilation
+        dnl scenarios).  So extract the values we want by using the CPP.
+        dnl
+        dnl See comment above about why we call this macro twice, once
+        dnl with nothing and once with -P.
+        _OPAL_CHECK_OFI_GET_API_VERSION([])
+        AS_IF([test -z "$OFI_MAJOR_VERSION" || test -z "$OFI_MINOR_VERSION"],
+              [_OPAL_CHECK_OFI_GET_API_VERSION([-P])])
+
+        dnl Sanity check: make sure we got values
+        AS_IF([test -z "$OFI_MAJOR_VERSION" || test -z "$OFI_MINOR_VERSION"],
+              [AC_MSG_RESULT([unknown])
+               AC_MSG_WARN([unable to run the C preprocessor])
+               AC_MSG_ERROR([Unable to continue])],
```

Comment:
It feels very weird to error here, because this is a test that might be unnecessary (if the user is only building the OFI BTL, not the OFI MTL, for example).  I think at a high level, this all should be in its own macro that is called only by the OFI MTL.  Further, get rid of all this code and, since we don't actually care about the version, only that it's greater than some number, have a macro OFI_MIN_VERSION(1,5) which expands to something like:

```
AC_TRY_COMPILE([#include <ofi.h>
#include <rdma/fabric.h>],[
#if FI_VERSION_LT(FI_VERSION(FI_MAJOR_VERSION, FI_MINOR_VERSION), FI_VERSION($1, $2))
#error "Nopes on the version
#endif 
```

Reply:
Yes, that is better.  I'll update.

Reply:
@bwbarrett Done.  Can you re-review?